WordPress/Classes/Utility/WPUserAgent.m

```diff
@@ -35,17 +35,33 @@ - (void)setupUserAgent
     [[NSUserDefaults standardUserDefaults] registerDefaults:dictionary];
 }
 
-#pragma mark - WordPress User-Agent
+#pragma mark - Default and WordPress User-Agents
+
+- (NSString *)defaultUserAgent
+{
+    // Temporarily unset "UserAgent" from registered user defaults so that we
+    // always get the default value, independently from what's currently set as
+    // User-Agent
+    NSDictionary *originalRegisteredDefaults = [[NSUserDefaults standardUserDefaults] volatileDomainForName:NSRegistrationDomain];
+
+    NSMutableDictionary *tempRegisteredDefaults = [NSMutableDictionary dictionaryWithDictionary:originalRegisteredDefaults];
+    [tempRegisteredDefaults removeObjectForKey:WPUserAgentKeyUserAgent];
+    [[NSUserDefaults standardUserDefaults] registerDefaults:tempRegisteredDefaults];
+    
+    NSString *userAgent = [[[UIWebView alloc] init] stringByEvaluatingJavaScriptFromString:@"navigator.userAgent"];
+    NSAssert(userAgent != nil, @"User agent shouldn't be nil");
+    NSAssert(! [userAgent isEmpty], @"User agent shouldn't be empty");
+    
+    [[NSUserDefaults standardUserDefaults] registerDefaults:originalRegisteredDefaults];
+    
+    return userAgent;
+}
```

Comment:
> This method seems very thread unfriendly as it is right now.

It is, however I've assumed that `WPUserAgent` is only to be used with UIWebView which always runs on the main thread (as the rest of the UIKit's classes). Said class exposes `useDefaultUserAgent` and `useWordPressUserAgent` which only have effect on UIWebView and not on other means of using HTTP.

It's possible that someone in the future might want to make `defaultUserAgent` public and use it directly for whatever purposes (e.g. REST calls), in which case I can go about wrapping parts of the method with `@synchronized`, but I'm not sure if you need it right now.

> Also, how does the addition of this method affect the following methods? Are they still safe to use?

The new method temporarily resets the UA, reads it, and sets it back. This wasn't being done previously, so I'd argue that `useWordPressUserAgent` was less safe before as it could have led to custom UA duplicating itself:

1. Class 1 creates WPUserAgent object, WPUserAgent reads default user agent, stores it to NSUserDefaults.
2. Class 1 calls `useWordPressUserAgent`, WPUserAgent sets UA to the custom value
3. Class 1 "forgets" to reset the UA to the default value
4. Class 2 creates WPUserAgent object, WPUserAgent reads what it thinks is a default user agent value but is actually a leftover custom value from class 1
5. If class 2 calls `useDefaultUserAgent`, custom UA instead of the default one is set

This wasn't actually happening in the app because WPUserAgent's users were well behaved, but the unit test was actually testing this specific erroneous behavior by setting UA to the default one and then comparing it to a custom UA (fixed in 0a41c20).